plugins/pulp_rpm/plugins/importers/yum/purge.py

```diff
@@ -286,3 +287,106 @@ def remove_unit_duplicate_nevra(unit, repo):
                                                             units_q=Q_nevra_filter,
                                                             yield_content_unit=True)
     repo_controller.disassociate_units(repo, unit_iterator)
+
+
+def remove_repo_duplicate_nevra(repo_id):
+    """
+    Removes duplicate units that have same NEVRA from a repo, keeping only the most recent unit
+
+    This function is for bulk operations on an entire repo, such as after syncing a repo.
+    When operating on single units, consider using :py:func:`remove_unit_duplicate_nevra` instead.
+
+    :param repo_id: ID of the repo from which units will be unassociated
+    :type repo_id: str
+    """
+    for unit_type in (models.RPM, models.SRPM, models.DRPM):
+        for unit_ids in _duplicate_key_id_generator(unit_type):
+            # q objects don't deal with order_by, so they can't be used with repo_controller funcs
+            # disassociate_units only uses the unit_id, so limit the resultset to only that field
+            rcus = model.RepositoryContentUnit.objects.filter(
+                repo_id=repo_id, unit_id__in=unit_ids).order_by('-updated').only('unit_id')
+
+            # 0 or 1 packages from the duplicate nevra search match this repo means no duplicates
+            if rcus.count() < 2:
+                continue
+
+            repo = model.Repository.objects.get(repo_id=repo_id)
+
+            # Since the repo_units queryset is ordered by the updated field (descending), the
+            # first repo content unit is the latest. All other RCUs should be disassociated
+            duplicate_units = (unit_type(id=rcu.unit_id) for rcu in rcus[1:])
+            repo_controller.disassociate_units(repo, duplicate_units)
+
+
+def _duplicate_key_id_generator(unit):
+    """duplicate NEVRA unit ID generator
+
+    :param unit: The unit whose NEVRA should be removed
+    :type unit: ContentUnit
```

Comment:
We usually put these at the end of the docblock, but I don't know if that matters.